### PR TITLE
feat: make plugin showDiff.jelly common for all pages

### PR DESCRIPTION
**Pull request by DuMaM**
This PR fixes a few small bugs connected with diffs and prepares code for upcoming changes.

Fixes and improvements:
* broken permissions in some usecases [JENKINS-69633](https://issues.jenkins.io/browse/JENKINS-69633)
* add missing `ShowDiff` button in history.jelly removed with https://github.com/jenkinsci/job-config-history-plugin/pull/255
* exctract common code for `ShowDiff` to component folder and reuse it in project files
* update pom with correct Jenkins versions

Preparations:
* I would like to add code mirror for diffs. [JENKINS-69069](https://issues.jenkins.io/browse/JENKINS-69069)

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by DuMaM** [comment]
@NotMyFault I would be grateful for suggestions.

**Review by NotMyFault — COMMENTED** [review]
Also there's a layout issue in XML according to the automated tests

File: src/main/resources/components/show-diff.jelly [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+
+                      document.getElementById("showHideVersionDiffsJsButton").value = hideText;
+                  } else {
+                      document.getElementById("tbody_versionDiffsShown").style.display = 'none';
+                      document.getElementById("tbody_versionDiffsHidden").style.display = '';
+
+                      document.getElementById("showHideVersionDiffsJsButton").value = showText;
+                  }
+              }
+          </script>
```
[the remark this answers is not in the document]
Comment (DuMaM) [reply src/main/resources/components/show-diff.jelly:83]: I just moved them :wink:
I will try to do something about it then.

**Comment by DuMaM** [comment]
@NotMyFault what are your thoughts on bringing this https://github.com/jenkinsci/pipeline-config-history-plugin inside this plugin?

**Comment by NotMyFault** [comment]
> @NotMyFault what are your thoughts on bringing this https://github.com/jenkinsci/pipeline-config-history-plugin inside this plugin? I would try to integrate how diff works and then maybe provide other features.

Do you think that is realistic? I haven't used the pipeline one yet, and I don't know which shape it is in...

**Comment by DuMaM** [comment]
> > @NotMyFault what are your thoughts on bringing this https://github.com/jenkinsci/pipeline-config-history-plugin inside this plugin? I would try to integrate how diff works and then maybe provide other features.
> 
> Do you think that is realistic? I haven't used the pipeline one yet, and I don't know which shape it is in...

There are parts of code which could be reused. I played with it a few months ago and it was fine.
I will bring file diffs first.

**Comment by cpfeiffer** [comment]
> add missing ShowDiff button in history.jelly removed with https://github.com/jenkinsci/job-config-history-plugin/pull/255

Is there an issue where the missing "Show diff" button can be tracked? I'm a bit confused about where it was lost, because in the "before" screenshot of #255, the diff button is already missing.

**Comment by DuMaM** [comment]
> > add missing ShowDiff button in history.jelly removed with #255
> 
> Is there an issue where the missing "Show diff" button can be tracked? I'm a bit confused about where it was lost, because in the "before" screenshot of #255, the diff button is already missing.

there is no `show diff` because other feature connected with file selection also didnt worked.
Still it didnt show when it was removed.

**Pull request closed without merging**